react/OrderPayment.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { createContext, ReactNode, useContext } from 'react'
+import { useQuery, useMutation } from 'react-apollo'
+import { cardSessionId as CardSessionIdQuery } from 'vtex.checkout-resources/Queries'
+import {
+  savePaymentToken as SavePaymentTokenMutation,
+  saveCards as SaveCardsMutation,
+} from 'vtex.checkout-resources/Mutations'
+
+interface Address {
+  postalCode: string
+  street: string
+  neighborhood: string
+  city: string
+  state: string
+  country: string
+  number: string
+  complement: string
+}
+
+interface PaymentData {
+  paymentSystem: string
+  cardHolder: string
+  cardNumber: string
+  expiryDate: string
+  csc: string
+  document: string
+  documentType: string
+  partnerId: string
+  address: Address
+}
+interface PaymentToken {
+  creditCardToken: string
+  paymentSystem: string
+}
+
+interface Status {
+  error: boolean
+  value: PaymentToken[] | string
+}
+interface Context {
+  savePaymentData: (paymentData: PaymentData[]) => Promise<Status>
+}
+
+interface OrderPaymentProps {
+  children: ReactNode
+}
+
+const OrderPaymentContext = createContext<Context | undefined>(undefined)
+
+export const OrderPaymentProvider = ({ children }: OrderPaymentProps) => {
+  const { data: cardSessionData, refetch } = useQuery(CardSessionIdQuery)
+
+  const [saveCardsMutation] = useMutation(SaveCardsMutation)
+  const [savePaymentTokenMutation] = useMutation(SavePaymentTokenMutation)
+
+  const savePaymentData = async (
+    paymentData: PaymentData[]
+  ): Promise<Status> => {
+    try {
+      const {
+        data: { saveCards },
```

Comment:
It shouldn't be save**d**Cards? 🤔 What is it exactly?

Reply:
The mutation return is like: 
```
data: {
  saveCards: {
    value
  }
}
```